okteto/okteto
# DEV-828: Don't allow execution of empty tests names

# Proposed changes

DEV-828

Don't allow to execute empty test containers names. We were not filtering empty strings as part of the arguments, so when an empty strings comes (for example, from the GitHub action), it is failing with the error `Don't know what to do with ""`.

So, we skip empty strings

This part of the code doesn't have unit tests, and I was changing integration tests, but I realized we are not running them never, so I will need to perform more changes. In order to have this fixed as soon as possible I will send a follow up PR, but I want to include this in the `3.2` release branch

## How to validate

Having the following manifest:

```
test:
  unit:
    commands:
      - name: hi
        command: echo "hi unit!"
  integration:
    commands:
      - name: hi
        command: echo "hi integration!"
```

1. Before these changes, execute `okteto test ""`, you will get the error
2. Then, compile this branch and execute the same command, you will see how all the tests are executed
3. Verify you can still execute individual or specific tests. Verify you can pass more than one test container name
